ui/src/views/infra/zone/ZoneWizardLaunchZone.vue

```diff
@@ -2093,7 +2093,11 @@ export default {
           resolve()
         }).catch(error => {
           message = error.response.headers['x-description']
-          reject(message)
+          if (message.includes('is already in the database')) {
```

Comment:
@weizhouapache Is this message coming for API response? Can check/correct the msg if required.

Reply:
yes @sureshanaparti 
it comes from the error message in API response.

```
plugins/hypervisors/hyperv/src/main/java/com/cloud/hypervisor/hyperv/discoverer/HypervServerDiscoverer.java:                logger.debug("Skipping " + agentIp + " because " + guidWithTail + " is already in the database.");
plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/discoverer/XcpServerDiscoverer.java:                    logger.debug("Skipping " + record.address + " because " + record.uuid + " is already in the database.");
server/src/main/java/com/cloud/hypervisor/kvm/discoverer/LibvirtServerDiscoverer.java:                        final String msg = "Skipping host " + agentIp + " because " + guid + " is already in the database for resource " + existingHost.getGuid() + " with ID " + existingHost.getUuid();
```

Reply:
Ok, if already there, maybe do have to check any other param changes for that host and update?

Reply:
@sureshanaparti 
do you mean the param passed in the wizard and the param of the existing host ?

Reply:
If already in database and added again, maybe we can show proper msg, something like Hosts already exists, re-added to the cluster.

Reply:
@sureshanaparti 
if host already exists, should we just ignore the error and skip the step ?

Reply:
as discussed, better update the backend msg, so that it'll be in sync with ui and cmk.

Reply:
@sureshanaparti 
I tested the following change (similar as others)
```
--- a/server/src/main/java/com/cloud/hypervisor/kvm/discoverer/LibvirtServerDiscoverer.java
+++ b/server/src/main/java/com/cloud/hypervisor/kvm/discoverer/LibvirtServerDiscoverer.java
@@ -247,7 +247,7 @@ public abstract class LibvirtServerDiscoverer extends DiscovererBase implements
                     if (existingHost.getGuid().toLowerCase().startsWith(guid.toLowerCase())) {
                         final String msg = "Skipping host " + agentIp + " because " + guid + " is already in the database for resource " + existingHost.getGuid() + " with I
+D " + existingHost.getUuid();
                         logger.debug(msg);
-                        throw new CloudRuntimeException(msg);
+                        return null;
                     }
                 }
             }
```

unfortunately I still got an exception 
```
2024-06-25T10:36:29,785 INFO  [c.c.r.ResourceManagerImpl] (qtp1449605932-20:[ctx-7faf5626, ctx-697f7fe4, ctx-dc5eb50d]) (logid:94985073) Trying to add a new host at http://10.0.35.82 in data center 1
2024-06-25T10:36:29,789 DEBUG [c.c.h.k.d.KvmServerDiscoverer] (qtp1449605932-20:[ctx-7faf5626, ctx-697f7fe4, ctx-dc5eb50d]) (logid:94985073) Skipping host 10.0.35.82 because 5ba056d0-0318-3ccc-93a7-037cf61305e2 is already in the database for resource 5ba056d0-0318-3ccc-93a7-037cf61305e2-LibvirtComputingResource with ID e7f4f60c-76bd-4924-8f07-d41baee5e29a
2024-06-25T10:36:29,789 WARN  [c.c.r.ResourceManagerImpl] (qtp1449605932-20:[ctx-7faf5626, ctx-697f7fe4, ctx-dc5eb50d]) (logid:94985073) Cannot find the server resources at http://10.0.35.82
2024-06-25T10:36:29,791 INFO  [c.c.u.e.CSExceptionErrorCode] (qtp1449605932-20:[ctx-7faf5626, ctx-697f7fe4, ctx-dc5eb50d]) (logid:94985073) Could not find exception: com.cloud.exception.DiscoveryException in error code list for exceptions
2024-06-25T10:36:29,792 WARN  [o.a.c.a.c.a.h.AddHostCmd] (qtp1449605932-20:[ctx-7faf5626, ctx-697f7fe4, ctx-dc5eb50d]) (logid:94985073) Exception: com.cloud.exception.DiscoveryException: Unable to add the host: Cannot find the server resources at http://10.0.35.82
    at com.cloud.resource.ResourceManagerImpl.discoverHostsFull(ResourceManagerImpl.java:894)
    at com.cloud.resource.ResourceManagerImpl.discoverHosts(ResourceManagerImpl.java:658)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
    at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344)
```

need some  investigation